pontoon/terminology/models.py

```diff
@@ -1,7 +1,44 @@
 import re
 
+from bulk_update.helper import bulk_update
+
 from django.db import models
 
+from pontoon.base.models import Entity, ProjectLocale, Resource, TranslatedResource
+
+
+def update_terminology_project_stats():
+    resource = Resource.objects.get(project__slug="terminology")
+    project = resource.project
+    total_strings = Entity.objects.filter(resource=resource, obsolete=False).count()
+    resource.total_strings = total_strings
+    resource.save(update_fields=["total_strings"])
+
+    translated_resources = TranslatedResource.objects.filter(resource=resource)
```

Comment:
As you get a list anyway, I'd get the list here. Then you don't rely on the query object caching it for you when you use the list in `bulk_update`.